src/core/qgsdatetimestatisticalsummary.cpp

```diff
@@ -66,6 +67,18 @@ void QgsDateTimeStatisticalSummary::addValue( const QVariant& value )
     testDateTime( date.isValid() ? QDateTime( date, QTime( 0, 0, 0 ) )
                   : QDateTime() );
   }
+  else if ( value.type() == QVariant::Time )
+  {
+    mIsTimes = true;
```

Comment:
Basically if one or more time values are passed (with no date) then this flags that the results should be times alone. Otherwise they'll be meaningless. It's cleared if the stats are reset.